### ISSUE TITLE
Degrade gracefully for non-ElastiCache backend

### ISSUE THREAD

**Issue by hectcastro** [gusdan/django-elasticache#16]
When using the `django_elasticache.memcached.ElastiCache` backend in a local development environment, failure to get a list of nodes from `config get cluster` throws an exception:

```
django_1     |   File "/usr/local/lib/python2.7/site-packages/django_elasticache/memcached.py", line 70, in get_cluster_nodes
django_1     |     get_cluster_info(server, port)['nodes'])
django_1     |   File "/usr/local/lib/python2.7/site-packages/django_elasticache/cluster_utils.py", line 43, in get_cluster_info
django_1     |     res = client.read_until(b'\n\r\nEND\r\n')
django_1     |   File "/usr/local/lib/python2.7/telnetlib.py", line 294, in read_until
django_1     |     return self._read_until_with_poll(match, timeout)
django_1     |   File "/usr/local/lib/python2.7/telnetlib.py", line 318, in _read_until_with_poll
django_1     |     else 1000 * call_timeout)
django_1     |   File "/usr/local/lib/python2.7/site-packages/gevent/select.py", line 127, in poll
django_1     |     result.event.wait(timeout=timeout)
```

We are currently dealing with this by conditionally setting the backend (which works well):

```python
if DEBUG:
    CACHE_BACKEND = 'django.core.cache.backends.memcached.PyLibMCCache'
else:
    CACHE_BACKEND = 'django_elasticache.memcached.ElastiCache'
```

It would be great if this plugin behaved like the base `PyLibMCCache` backend when a non-ElastiCache endpoint is provided.

**Comment by gusdan** [comment]
I'm not sure if your approach is correct. CACHE_BACKEND very old django settings, currently django supports cache configuration like:

```
CACHES = {
    'default': {
        'BACKEND': 'django_elasticache.memcached.ElastiCache',
        'LOCATION': 'xxxxxxx.use1.cache.amazonaws.com:11211',
    }
}
```

in local settings it can be replaced with


```
CACHES = {
    'default': {
        'BACKEND': 'django.core.cache.backends.memcached.PyLibMCCache',
        'LOCATION': 'localhost:11211',
    }
}
```

Anyway production and local environments cannot have same "LOCATION". So i don't see usecase where we should allow django_elasticache to connect to memcache. I.e. you either override all cache settings (location and backend) or keep them same.

Also i'm not sure about default timeout of 3 seconds. For example on production we have very big memcache timeout, during peak load polling (select) can wait a while (more than 3 seconds) and with your fix elastic cache will switch to use one node (instead of all nodes in cluster) and we will have even worse performance in bad moment.

I.e. i see such bad scenario:
1. We have problems on server (high load)
2. Our server cannot perform all request (100% CPU, bad network)
3. We restart worker (gunicorn or someone else)
4. Worker starts cluster discovery with 3 seconds timeout
5. Since we don't have enough resources on server we didn't get response in 3 seconds
6. We run worker with configuration where we have just one default worker
7. Caching layer of site became not effective
8. We have more job for main app and our crisis becomes even worse

**Comment by hectcastro** [comment]
> I'm not sure if your approach is correct. CACHE_BACKEND very old django settings, currently django supports cache configuration like:

This was more of a poor choice in variable name. In our `settings.py`, the example above uses the `CACHE_BACKEND` variable to set the `CACHES` dict:

```python
CACHES = {
    'default': {
        'BACKEND': 'django.core.cache.backends.locmem.LocMemCache'
    },
    # Use a custom separate cache for custom DRF throttling classes, since these
    # don't require a heavyweight cache
    'api_throttling': {
        'BACKEND': CACHE_BACKEND,
        'LOCATION': CACHE_LOCATION
    }
}
```

> Anyway production and local environments cannot have same "LOCATION". So i don't see usecase where we should allow django_elasticache to connect to memcache. I.e. you either override all cache settings (location and backend) or keep them same.

In our case they can because we reference the cache endpoint using a private DNS record, for example `cache.service.app.internal`. When the application comes up locally, work it done to have that DNS record resolve to a local Memcached instance. When the application comes up in AWS, we use Route 53 private hosted zones to point to ElastiCache.

> Also i'm not sure about default timeout of 3 seconds. For example on production we have very big memcache timeout, during peak load polling (select) can wait a while (more than 3 seconds) and with your fix elastic cache will switch to use one node (instead of all nodes in cluster) and we will have even worse performance in bad moment.

Are you suggesting that I thread through the Memcached client timeout, or take an alternate approach to switching between the two types of endpoints? I agree that the timeout value is arbitrary. Maybe something like an additional boolean setting that can be used to determine if you're targeting a single node, or a discovery endpoint aware cluster?

**Comment by gusdan** [comment]
Yes i see. Maybe instead of timeout we can parse output of "cluster" command without waiting for b'\n\r\nEND\r\n' and if we get ERROR we will switch to base endpoint.

Also i like your idea with "additional boolean setting", what do you think about SUPPORT_MEMCAHED? or   IGNORE_CLUSTER_ERRORS?

**Comment by hectcastro** [comment]
I updated the associated PR to make use of `Telnet.expect()` so that it can match against the success (`END`) and failure case (`ERROR`) at the same time. I think that this achieves your desired behavior of immediately matching against either case.

If that looks good, I'll look into adding support for the boolean setting.

**Comment by gusdan** [comment]
I added few comments to your pull request. But in general it looks good, so you can add special settings, remove timeouts and we will create new version

**Comment by hectcastro** [comment]
I added the `IGNORE_CLUSTER_ERRORS` option and an additional test to cover when this option is enabled or disabled.

Unfortunately, I don't see any of your PR comments. The conversation count for https://github.com/gusdan/django-elasticache/pull/17 is still at zero for me.

**Comment by gusdan** [comment]
@hectcastro thanks for your contribution. I just released 1.0.1 version.
